src/Components/WebAssembly/WebAssembly/src/Infrastructure/JSInteropMethods.cs

```diff
@@ -29,4 +29,13 @@ public static void NotifyLocationChanged(string uri, string? state, bool isInter
     {
         WebAssemblyNavigationManager.Instance.SetLocation(uri, state, isInterceptedLink);
     }
+
+    /// <summary>
+    /// For framework use only.
+    /// </summary>
+    [JSInvokable(nameof(NotifyLocationChanging))]
+    public static void NotifyLocationChanging(int callId, string uri, bool isInterceptedLink)
+    {
+        WebAssemblyNavigationManager.Instance.HandleLocationChanging(callId, uri, isInterceptedLink);
+    }
```

Comment:
Ah, I think this is the reason. These APIs target the static fields/values.

Maybe we need to file an issue here to see if we could convert some of these things to use JSObjectReferences for dispatching

Reply:
This is something I thought about while implementing this feature - I wonder we could simplify things if we had a `DotNetObjectReference` to some object responsible for invoking the "location changing" handlers. Then we wouldn't have to separate out things into "notify location changing" and "end location changing" steps since a similar mechanism is handled behind the scenes in the `JSRuntime`. But maybe there's some reason why this wouldn't work. Is this approach what you had in mind?

Reply:
@MackinnonBuck something like that might work. I would suggest you file an issue and give it a try in the future if you want if you have spare cycles. (After RC1)